icicle/include/icicle/sumcheck/sumcheck_transcript_config.h

```diff
@@ -0,0 +1,113 @@
+#pragma once
+
+#include "icicle/hash/hash.h"
+
+namespace icicle {
+
+  /**
+   * @brief Configuration for encoding and hashing prover messages in the Sumcheck protocol.
+   *
+   * This class describes how prover messages are encoded and hashed to deterministically
+   * generate randomness. It includes:
+   * - Labels for various protocol stages (e.g., domain separator, round polynomials, challenges).
+   * - Endianness for encoding messages (little-endian by default).
+   * - The hashing function used to derive randomness.
+   * - A seed for initializing the random number generator (RNG).
+   *
+   * @tparam S Type of the field element (e.g., prime field or extension field elements).
+   */
+  template <typename S>
+  class SumcheckTranscriptConfig
+  {
+  private:
```

Comment:
I prefer placing the private section at the bottom

Reply:
ok I can do that